adapters/improvedigital/improvedigital.go

```diff
@@ -144,3 +147,77 @@ func getMediaTypeForImp(impID string, imps []openrtb2.Imp) (openrtb_ext.BidType,
 		Message: fmt.Sprintf("Failed to find impression for ID: \"%s\"", impID),
 	}
 }
+
+// This method responsible to clone request and convert additional consent providers string to array when additional consent provider found
+func (a *ImprovedigitalAdapter) applyAdditionalConsentProvidersArray(request openrtb2.BidRequest) openrtb2.BidRequest {
+	// If user/user.ext not defined, no need to parse additional consent
+	if request.User == nil || request.User.Ext == nil {
+		return request
+	}
+
+	var (
+		err     error
+		reqJSON json.RawMessage
+		cpStr   string
+	)
+
+	// Start validating additional consent
+	// Check key exist user.ext.ConsentedProvidersSettings.consented_providers
+	var userExtMap = make(map[string]json.RawMessage)
+	if err = json.Unmarshal(request.User.Ext, &userExtMap); err != nil {
```

Comment:
Is there a scenario where we would want to handle the error here differently than just returning the request?

Reply:
Could you also add a test that triggers this error case here?

Reply:
applyAdditionalConsentProvidersArray method supposes to handle only additional consent-related logic. If it can't parse necessary data, it just returns the original request and don't interfere with request.

PBS already validates request data (as like if user.ext assigned a string or an array then validation error occurs). So usually it won't go into the return statement (line: 168). but we add an error handler here to avoid any surprise error.